syntax = "proto3";
import "google/protobuf/empty.proto";

service WorkerService {
    rpc StartStep(StartStepRequest) returns (StartStepReply) {}
    rpc WorkerStatus(google.protobuf.Empty) returns (WorkerStatusReply) {}
}

message StartStepRequest {
    string inputLocation = 1;
    string outputLocation = 2;
    string stepId = 3;
    // NOTE: start is inclusive, end is exclusive
    int64 rangeStart = 4;
    int64 rangeEnd = 5;
    string workerCodeLocation = 6;
}

message StartStepReply {
    bool ok = 1;
    string workerUuid = 2;
}

message WorkerStatusReply {
    enum WorkerStatusEnum {
        Unspecified = 0;
        Ok = 1;
        Working = 2;
        Failure = 3;
    }
    WorkerStatusEnum status = 1;
    string errorMessage = 2;
    string workerUuid = 3;
}
